pi4j-core/src/main/java/com/pi4j/io/gpio/digital/DigitalOutputBase.java

```diff
@@ -128,10 +128,100 @@ public Future<?> pulseAsync(int interval, TimeUnit unit, DigitalState state, Cal
     }
 
     /** {@inheritDoc} */
+    /**
+     * This method will blink an output pin of the RPi according the given specifications.
+     * The pin itself is created while creating a DigitalOutput configuration where one of
+     * the parameters is an address (= a BCM pin number).
+     *
+     * @param delay - The toggle time.
+     * @param duration - The amount of times the output has to toggle.
+     *
+     * Representation:
+     *
+     *   HIGH +-----+     +-----+     +-----+
+     *        |     |     |     |     |     |
+     *   LOW  +     +-----+     +-----+     +-----+
+     *        ^                                   ^
+     * start -┘                                   └- stop
+     *         \___/ \___/
+     *         delay  delay
+     *
+     *        \___________________________________/
+     *                      duration
+     *
+     * Example: Delay = 1 sec / duration = 10
+     *          Output will be like so (suppose the initial state is ON):
+     *          1 - 0 - 1 - 0 - 1 - 0 - 1 - 0 - 1 - 0 with each state lasting for 1 second.
+     *          So, if you would connect a LED to the pin, you would see the LED switching
+     *          on and off for 5 times, NOT 10 times!!!
+     *
+     * @param unit - The time unit used to calculate the delay.
+     * @param state - The initial state of the pin.
+     * @param callback - The method to call, if any, once the blinking is done.
+     * @return - The DigitalOutput object itself.
+     */
     @Override
     public DigitalOutput blink(int delay, int duration, TimeUnit unit, DigitalState state, Callable<Void> callback) {
-        // TODO :: IMPLEMENT DIGITAL OUTPUT BLINK
-        throw new UnsupportedOperationException("BLINK has not yet been implemented!");
+        int millis = 0;
+
+        if (delay <= 0) {
+            throw new IllegalArgumentException("A delay of zero or less is not supported.");
+        }
+        if (duration <= 0) {
+            throw new IllegalArgumentException("A duration of zero or less is not supported.");
+        }
+
+        switch (unit) {
+            case MICROSECONDS:
+                throw new IllegalArgumentException("TimeUnit.MICROSECONDS is not supported.");
+
+            case MILLISECONDS:
+                millis = delay;
+                break;
+
+            case SECONDS:
+                millis = (delay * 1000);
+                break;
+
+            case MINUTES:
+                millis = (delay * 1000 * 60);
+                break;
+
+            case HOURS:
+                millis = (delay * 1000 * 60 * 60);
+                break;
+
+            default:
+                throw new IllegalArgumentException("TimeUnit provided is not supported.");
+        }
+
+        this.state(state);
+
+        for (int i = 0; i < duration; i++) {
+            // block the current thread for the pulse duration
+            try {
+                Thread.sleep(millis);
```

Comment:
Performing a Thread.sleep works, but it makes this method blocking. We should rather use a ScheduledExecutor which can perform this task for us and makes this method non-blocking, which i am sure is mostly desired.

Reply:
ok, i just saw that we have a pulse() and a pulseAsync, so we can make this method blocking, but then we need to implement the async method as well...

Reply:
Indeed, I wrote you a personal e-mail telling you there was a blink() and blinkAsync(), just like there is a pulse() and pulseAsync() method already available.  
I've kept both blink() and pulse() methods as blocking methods and implemented the blinkAsync() and pulseAsync() methods with the ScheduledExecutorService.  Used a scheduleAtFixedRate() call for the blinkAsync() and a schedule() for the pulseAsync() method.